lib/hdt.js

```diff
@@ -36,6 +36,25 @@ HdtDocumentPrototype.searchLiterals = function (substring, options, callback, se
   this._searchLiterals(substring, offset, limit, callback, self);
 };
 
+
+// Searches the document for literals that contain the given string
+HdtDocumentPrototype.findTerms = function (options, callback, self) {
+  if (typeof  callback !== 'function') return;
+  if (this.closed) return callback.call(self || this, new Error('The HDT document cannot be read because it is closed'));
+  var limit  = options && options.limit  ? Math.max(0, parseInt(options.limit,  10)) : 100;
+  var position = options && options.position  ? options.position.toLowerCase() : undefined;
+  var prefix = options && options.prefix ? options.prefix : '';
+  var posId;
+  switch (position) {
```

Comment:
Let's not do this with a `switch`, but rather with a constant `POSITIONS` object declared in top scope. I.e., `var posId = POSITIONS[position]`.